test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -114,6 +119,12 @@ class test_ucp_worker_discard : public ucp_test {
             }
         } while (m_destroyed_ep_count < m_created_ep_count);
 
+        if (UCS_PTR_IS_PTR(flush_nb_req)) {
+            status = ucp_request_check_status(flush_nb_req);
+            EXPECT_UCS_OK(status);
+            ucp_request_release(flush_nb_req);
+        }
```

Comment:
we should check flush completion in the do/while loop
and after loop check all eps are destroyed

Reply:
done, checking `ucp_worker_t::flush_ops_count` field as discussed offline